sherpa/astro/io/tests/test_io_pha.py

```diff
@@ -0,0 +1,1366 @@
+
+    outpha = io.read_pha(str(outfile))
+    assert isinstance(outpha, DataPHA)
+    assert outpha.name.endswith("/test.pha")
+    check_header(outpha)
+    assert outpha.header["HDUCLAS4"] == "TYPE:I"
+    check_data(outpha)
+
+    # The responses and background should NOT be read in
+    # (we are in a different directory to infile so we can't
+    # find these files).
+    #
+    assert outpha.response_ids == []
+    assert outpha.background_ids == []
+
+    assert outpha.get_arf() is None
+    assert outpha.get_rmf() is None
+    assert outpha.get_background() is None
+
+    if backend_is("crates"):
+        check_csc_pha_roundtrip_crates(outfile)
+
+    elif backend_is("pyfits"):
+        check_csc_pha_roundtrip_pyfits(outfile)
+
+    else:
+        raise RuntimeError(f"Unknown io backend: {io.backend}")
+
+
+@pytest.mark.parametrize("column", ["channel", "counts"])
+@pytest.mark.parametrize("is_ascii", [True, False])
+@requires_fits
+def test_write_pha_missing_column(column, is_ascii, tmp_path):
+    """What happens when there's no "required" column.
+
+    At the moment we write out the file.
+    """
+
+    chans = np.arange(1, 4, dtype=np.int32)
+    counts = np.arange(1, 4, dtype=np.int32)
+    pha = DataPHA("tmp", chans, counts)
+    setattr(pha, column, None)
+
+    tmpfile = tmp_path / 'test.out'
+    assert not tmpfile.exists()
+    io.write_pha(str(tmpfile), pha, ascii=is_ascii)
+
+    # We want to check we've created a single-column file, but
+    # it depends on the backend and whether it was written
+    # out in ASCII or FITS format. As we don't really care here,
+    # all we do is check the output file exists and is not empty.
+    #
+    assert tmpfile.exists()
+    assert tmpfile.is_file()
+    assert tmpfile.stat().st_size > 0
+
+
+@requires_fits
+def test_write_pha_columnn_length_mismatch_ascii(tmp_path):
+    """What happens when "required" column do not match.
+
+    This is separate from test_write_pha_columnn_length_mismatch_fits
+    as the behavior is different-enough it's not worth adding a
+    is_ascii parameter. It also depends on the backend.
+    """
+
+    chans = np.arange(1, 4, dtype=np.int32)
+    counts = np.arange(1, 6, dtype=np.int32)
+    pha = DataPHA("tmp", chans, counts)
+
+    tmpfile = tmp_path / 'test.fits'
+    if backend_is("crates"):
+        io.write_pha(str(tmpfile), pha, ascii=True)
+        dat = io.read_table(str(tmpfile))
+        assert len(dat.x) == 5
+        assert len(dat.y) == 5
+        assert dat.x == pytest.approx([1, 2, 3, 0, 0])
+        assert dat.y == pytest.approx([1, 2, 3, 4, 5])
+
+    elif backend_is("pyfits"):
+        with pytest.raises(IOErr) as err:
+            io.write_pha(str(tmpfile), pha, ascii=True)
+
+        assert str(err.value) == "not all arrays are of equal length"
+
+    else:
+        assert False  # programming error
+
+
+@requires_fits
+def test_write_pha_columnn_length_mismatch_fits(tmp_path):
```

Comment:
I understand that you are just adding tests for existing behavior, but that behavior does not make any sense. Are you planning to address this or do we have an open issue for it we can reference here?

Reply:
I have this in my mind under the generic "we don't validate the Data fields" area. If you want to make an issue of it then please go ahead but I don't have time right now.

Reply:
See #1065